### ISSUE TITLE
Upgrading to Drupal 6.38 - handling of unsupported projects

### ISSUE THREAD

**Issue by hansfn** [drush-ops/drush#2024]
```
$ drush up drupal
Update information last refreshed: tor, 2016-02-25 07:56
 Name    Installed Version  Proposed version  Message
 Drupal  6.37               6.37              Installed version not supported

No code updates available. 
```

but

```
$ drush rl drupal
 Project  Release  Date         Status
 drupal   6.x-dev  2016-Feb-24  Development
 drupal   6.38     2016-Feb-24  Security
```

Is there anyway to force Drush to upgrade an unsupported project? Or is the problem something else? (Drush 8.0.3 on Linux.)

**Comment by weitzman** [comment]
I think that's the issue, and i see no switch to force alternate behavior.

**Comment by kordianbruck** [comment]
There is a new issue in Drupal Core as well: https://www.drupal.org/node/2675892

**Comment by hansfn** [comment]
Thx for the pointer to the Drupal Core issue.

OK, so maybe the issue will be "fixed" without a new Drush release. I still think it's worth considering if there should be "a switch to force alternate behavior".

**Comment by kordianbruck** [comment]
@hansfn there should definitely be a "force update" status - I don't think much will happen on Drupals side. 

I'm happy to create a PR if someone can point me into the right direction.

**Comment by alexm** [comment]
It [seems](https://www.drupal.org/node/2675892#comment-10898788) that there may be an official solution soon:

> This was done as part of the EOL support. We have a meeting to chat about it how to fix it later on today.

**Comment by alexm** [comment]
I debugged our drush 6.2.0 (it's an old ubuntu server with an old php version) and finally forced the update adding this:

```
--- /usr/share/php/drush/commands/pm/updatestatus.pm.inc.orig	2016-02-26 16:42:04.176508542 +0100
+++ /usr/share/php/drush/commands/pm/updatestatus.pm.inc	2016-02-26 16:21:30.646442140 +0100
@@ -112,6 +112,8 @@
   // Build project updatable messages, set candidate version and mark
   // 'updatable' in the project.
   foreach ($update_info as $key => $project) {
+    // LCAC
+    $project['updateable'] = TRUE;
     switch($project['status']) {
       case DRUSH_PM_REQUESTED_UPDATE:
         $status = dt('Specified version available');
@@ -183,6 +185,8 @@
  * Set a release to a recommended version (if available), and set as updateable.
  */
 function pm_release_recommended(&$project) {
+  // LCAC
+  $project['recommended'] = '6.38';
   if (isset($project['recommended'])) {
     $project['candidate_version'] = $project['recommended'];
     $project['updateable'] = TRUE;
```

**Comment by greg-1-anderson** [comment]
I considered looking at this, but Drupal 6 won't install under my current php configuration.

It would be helpful if we had a Drupal 6 maintainer.

**Comment by weitzman** [comment]
Fixed by Drupal infra

**Comment by jonhattan** [comment]
@greg-1-anderson to maintain Drupal 6, we should start by enabling tests run on 6.x :smile:

**Comment by greg-1-anderson** [comment]
Yes, I was thinking the same thing!  Even if they do not pass, it would be super helpful if we had at least ONE of the test files running on Drupal 6.

"If it's not tested, it probably doesn't work." And even if it does happen to work, is it really supported?  We should get tests turned on for the Drush 8 branch before we split off Drush 9 (and maybe de-support Drupal 6 again in the Drush 9 branch).  I'll work on that when I get a moment, but I won't complain if someone else beats me to it.

**Comment by greg-1-anderson** [comment]
#2045

**Comment by greg-1-anderson** [comment]
Maybe not fully fixed by Drupal infrastructure?

From #2045:
```
1) Unish\commandCase::testDisabledModule
Unexpected exit code 75 (expected 0) for command:
/home/travis/build/drush-ops/drush/drush --root=/tmp/drush-sandbox/web --uri=dev --nocolor pm-download devel --cache
Command output:
Choose one of the available releases for devel:
 [0]  :  Cancel                                      
 [1]  :  6.x-1.x-dev  -  2014-Oct-23  -  Development 
 [2]  :  6.x-1.25     -  2011-Jul-20  -  Security
Command stderr:
There are no stable releases for project devel.                        [warning]
Cancelled.                                                              [cancel]
```

**Comment by weitzman** [comment]
Yeah, it wasn't fixed by Drupal infra. But it was fixed today by @jonhattan in #2042. Thats a slight behavior improvement when a branch is unsupported (which is true for all of D6 contrib except projects that actively choose to stay supported).
